src/cljs/harja/tiedot/hallinta/tyokalut/talvihoitoreitti_tyokalu.cljs

```diff
@@ -0,0 +1,251 @@
+                                 (assoc :valittu-urakka nil)))
+                             talvihoitoreitti)
+          ;; Aseta urakoitsija
+          talvihoitoreitti (if (and (:valittu-urakka talvihoitoreitti) (get-in talvihoitoreitti [:valittu-urakka :urakoitsija]))
+                             (assoc talvihoitoreitti :suorittaja-nimi (get-in talvihoitoreitti [:valittu-urakka :urakoitsija :nimi]))
+                             talvihoitoreitti)]
+      (-> app
+        (assoc :mahdolliset-urakat (if (and
+                                         @nav/hallintayksikon-urakkalista
+                                         (not= +mahdolliset-urakat+ @nav/hallintayksikon-urakkalista))
+                                     @nav/hallintayksikon-urakkalista
+                                     +mahdolliset-urakat+))
+        (assoc :talvihoitoreitti talvihoitoreitti))))
+
+  Laheta
+  (process-event [{talvihoitoreitti :talvihoitoreitti} app]
+    (let [tulos! (tuck/send-async! ->LahetysOnnistui)
+          virhe! (tuck/send-async! ->LahetysEpaonnistui)
+          urakkaid (:id (get-in app [:talvihoitoreitti :valittu-urakka]))]
+
+      (.setTimeout js/window
+        #(let [data (koostettu-data app)]
+           (if urakkaid
+             (go
+               (let [params {:body (.stringify js/JSON (clj->js data))
+                             :content-type :json
+                             :accept :json}
+                     ;; Lähetä talvihoitoreitti rajapintaan
+                     vastaus (<! (http/put (str "api/urakat/" urakkaid "/talvihoitoreitti") params))
+                     _ (println "Laheta :: vastaus" (pr-str vastaus))]
+                 (if (or (k/virhe? vastaus) (= 400 (:status vastaus)))
+                   (virhe!)
+                   (tulos!))))
+             (js/console.log "Urakkaa ei valittu! ei lähetetä (käyttöliittymä on vaikea, eli yritä uusiksi)")))
+
+        500)
+
+      (assoc app :lahetys-kaynnissa? true)))
+
+  LahetysOnnistui
+  (process-event [{vastaus :vastaus} app]
+    (do
+      (viesti/nayta-toast! "Talvihoitoreitti lähetetty" :onnistui)
+      (assoc app :lahetys-kaynnissa? false)))
+
+  LahetysEpaonnistui
+  (process-event [{vastaus :vastaus} app]
+    (do
+      (viesti/nayta-toast! "Talvihoitoreitin lähetys epäonnistui" :varoitus viesti/viestin-nayttoaika-pitka)
+      (assoc app :lahetys-kaynnissa? false)))
+
+  HaeHallintayksikonUrakatOnnistui
+  (process-event [{vastaus :vastaus} app]
+    ;; Filtteröidään muut, kuin teiden-hoito tyyppiset urakat pois. Koska talvihoitoreittejä ei voi muille lisätä
+    (let [vastaus (filter #(= :teiden-hoito (:tyyppi %)) vastaus)]
+      (viesti/nayta-toast! "HaeHallintayksikonUrakatOnnistui" :onnistui)
+      (assoc app :mahdolliset-urakat vastaus)))
+
+  HaeHallintayksikonUrakatEpaonnistui
+  (process-event [{vastaus :vastaus} app]
+    (js/console.log "HaeHallintayksikonUrakatEpaonnistui :: vastaus" (pr-str vastaus))
```

Comment:
Täällä aika paljon console logitusta, halutaanko näitä kaikkea täällä pitää? Vaikka onkin hallinnan puolen työkalu

Reply:
Näistä iso osa on vaan copypastea toteumatyökalun puolelta. Nämä on pelkästän vain meidän kehityskäyttöön, niin koodin hieronnalla ei saada vastinetta asiakkaan rahoille. En aio siivota näitä kommentteja pois tai refaktoroida tätä, ellen satu tekemään tuohon työkaluun jossain vaiheessa isommin muutoksia.